### PR TITLE
✨ Allow updating the title of the page using amp-bind

### LINKED ISSUE #14560: Intent to Implement: Changing <title> with amp-bind
We should allow AMP pages to change their titles dynamically by adding a `[text]` attribute to the `<title>` tag.

```
<head>
...
  <title [text]="myState.title">My AMP Page</title>
...
</head>
```

cc @choumx

### PR DESCRIPTION
**Pull request by josh313**
Fixes #14560 

@choumx

### CODE REVIEW

File: validator/validator-main.protoascii [hunk 1]
```diff
@@ -120,6 +120,7 @@ tags: {
 tags: {
   tag_name: "TITLE"
   spec_name: "title"
+  attrs: { name: "[text]" }
```
Comment (choumx) [inline validator/validator-main.protoascii]: @honeybadgerdontcare FYI validator change.

File: extensions/amp-bind/0.1/bind-impl.js [hunk 2]
```diff
@@ -282,14 +283,19 @@ export class Bind {
   /**
    * Scans the ampdoc for bindings and creates the expression evaluator.
    * @param {!Node} rootNode
+   * @param {!Node|null} titleNode
```
Comment (choumx) [inline extensions/amp-bind/0.1/bind-impl.js]: Nit: `@param {Node} titleNode`

File: extensions/amp-bind/0.1/bind-impl.js [hunk 3]
```diff
@@ -868,6 +874,9 @@ export class Bind {
     switch (property) {
       case 'text':
         element.textContent = String(newValue);
+        if (tag === 'TITLE') {
+          this.win_.document.title = String(newValue);
```
Comment (choumx) [inline extensions/amp-bind/0.1/bind-impl.js]: `this.localWin_` is important for the "many AMPs per page via shadow DOM" PWA case.

File: validator/validator-main.protoascii [hunk 4]
```diff
@@ -26,7 +26,7 @@ min_validator_revision_required: 322
 # newer versions of the spec file. This is currently a Google internal
 # mechanism, validator.js does not use this facility. However, any
 # change to this file (validator-main.js) requires updating this revision id.
-spec_file_revision: 602
+spec_file_revision: 603
```
Comment (honeybadgerdontcare) [inline validator/validator-main.protoascii]: No need to increment this with the change.

File: validator/validator-main.protoascii (hunk 1 above)
Comment (honeybadgerdontcare) [reply validator/validator-main.protoascii]: Validation change looks fine.

**Comment by honeybadgerdontcare** [comment]
@choumx 
Travis is failing with the following, which one of the failure is a test in this PR.

```
○○○○○○○○○○○○○○○○○○○○○○○○○○○○○○○○○○○○●●●○○○○○○○○○○○●●○○○○●○○●●●●●
DESCRIBE => Bind
  DESCRIBE => in shadow ampdoc
    DESCRIBE =>  
      IT => should scan for bindings when ampdoc is ready
        ✗ TypeError: element.getElementsByTagName is not a function
            at elementByTag (absolute/tmp/70e1f428d4d399a566d0f339d47dcd49.browserify?89a91f760b2df6dd9e6b7c274d3a80736b929b92:45702:26)
            at absolute/tmp/70e1f428d4d399a566d0f339d47dcd49.browserify?89a91f760b2df6dd9e6b7c274d3a80736b929b92:3114:60
●●●●●●●●●●●●●●
  DESCRIBE => in single ampdoc
    DESCRIBE =>  
      IT => should update document title for <title> elements
        ✗ AssertionError: expected 'foo' to equal 'abc'
            at absolute/tmp/70e1f428d4d399a566d0f339d47dcd49.browserify?89a91f760b2df6dd9e6b7c274d3a80736b929b92:5094:34
●●●●●●●●●●●●●●●●●●●●●●●●○●●○○○○○●○○○●●●●○○●○○●
Chrome 65.0.3325 (Windows 7 0.0.0): Executed 253 of 374 (Skipped 121) 2 FAILED
```

**Comment by choumx** [comment]
Investigating. Looks like it failed with `--compiled`.

**Pull request merged**